src/transformers/modeling_bart.py

```diff
@@ -823,6 +823,12 @@ def forward(
             output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
         )
 
+        if decoder_input_ids is None:
```

Comment:
@sshleifer - would be great if you can take a look here! After having though about it, I think it would be best to have `use_cache=True` as the default for Bart and disable it if not possible. 

It would break backward compatiiblity in a sense that before when just running the forward pass with `input_ids` and `decoder_input_ids`, the model would not use the cache, whereas now it does. However for v3.0.0 I think this is OK. 

Setting `config.use_cache=False` for Bart would mean that the generate method would also not use cache, which I think is the worse option of the two. 

The last option would be to just not touch Bart, but this would mean that the user cannot influence `use_cache` via `config.use_cache` (except for the generate method, but not for the forward pass), which will also lead to problems when making Bart torchscript ready.

Reply:
I completely agree with you!